pkg/reader/ccip.go

```diff
@@ -1322,6 +1309,89 @@ func (r *ccipChainReader) getOffRampSourceChainsConfig(
 	return configs, nil
 }
 
+// fetchFreshSourceChainConfigs always fetches fresh source chain configs directly from contracts
+// without using any cached values. Use this when up-to-date data is critical, especially
+// for sequence number accuracy.
+func (r *ccipChainReader) fetchFreshSourceChainConfigs(
+	ctx context.Context,
+	destChain cciptypes.ChainSelector,
+	sourceChains []cciptypes.ChainSelector,
+) (map[cciptypes.ChainSelector]SourceChainConfig, error) {
+	lggr := logutil.WithContextValues(ctx, r.lggr)
+
+	reader, exists := r.contractReaders[destChain]
+	if !exists {
+		return nil, fmt.Errorf("no contract reader for chain %d", destChain)
+	}
+
+	// Filter out destination chain
+	filteredSourceChains := filterOutChainSelector(sourceChains, destChain)
+	if len(filteredSourceChains) == 0 {
+		return make(map[cciptypes.ChainSelector]SourceChainConfig), nil
+	}
+
+	// Prepare batch requests for the sourceChains
```

Comment:
```suggestion
	// Prepare batch requests for the sourceChains to fetch the latest Unfinalized config values.
```

Reply:
You don't get to specify finalized/unfinalized for batch requests and I keep forgetting what the default is... worth calling out in comments I think.